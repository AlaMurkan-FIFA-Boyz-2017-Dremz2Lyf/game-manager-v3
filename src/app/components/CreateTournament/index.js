import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Container, Segment, Header, Message, Button, Modal } from 'semantic-ui-react';

class CreateTournament extends PureComponent {
  static propTypes = {
    tournament: PropTypes.bool,
  }

  state = {
    modalOpen: false,
  }

  closeModal = () => {
    this.setState({
      modalOpen: false,
    });
  }

  openModal = () => {
    this.setState({
      modalOpen: true,
    });
  }

  render() {
    const { modalOpen } = this.state;

    return (
      <Modal
        open={modalOpen}
        onOpen={this.openModal}
        onClose={this.closeModal}
        trigger={<Button icon="plus" content="New" />}
        basic
        size="small"
      >
        <Modal.Header>Select a Photo</Modal.Header>
        <Modal.Content>
          <Modal.Description>
            <Header>Default Profile Image</Header>
          </Modal.Description>
        </Modal.Content>
      </Modal>
    );
  }
}

export default CreateTournament;
