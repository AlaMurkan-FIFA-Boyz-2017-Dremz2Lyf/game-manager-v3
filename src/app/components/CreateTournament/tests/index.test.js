import { shallow } from 'enzyme';
import React from 'react';

import CreateTournament from '../index';

describe('<CreateTournament />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CreateTournament />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a closeModal method', () => {
    wrapper.setState({ modelOpen: true });
    wrapper.instance().closeModal();
    expect(wrapper.state({ modelOpen: false }));
  });

  it('should have a openModal method', () => {
    expect(wrapper.state({ modelOpen: false }));
    wrapper.instance().openModal();
    expect(wrapper.state({ modelOpen: true }));
  });
});
