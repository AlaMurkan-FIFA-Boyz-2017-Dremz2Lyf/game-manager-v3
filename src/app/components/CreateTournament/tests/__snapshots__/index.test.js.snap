// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateTournament /> should match snapshot 1`] = `"<button class=\\"ui button\\" role=\\"button\\"><i aria-hidden=\\"true\\" class=\\"plus icon\\"></i>New</button>"`;
